factory-reset/eos-factory-reset

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python3
+import argparse
+import sys
+import subprocess
+
+# Stages list
+# Each stage has the form (short option, name, description)
+# In addition from the stages listed bellow, the following stage names and
+# short option names are already taken by other command line arguments:
+# ('q', 'quiet', 'Suppress all output and don\'t ask for confirmation')
+# ('D', 'disable', 'Disable the selected stages instead of enabling them')
+STAGES = [
+    ('u', 'users', 'remove all user accounts (and home directory)'),
+]
+
+def yes_or_no(msg):
+    answer = ''
+    while answer not in ['y', 'n']:
+        answer = input(msg + ' ').lower()
+    return answer == 'y'
+
+if __name__ == '__main__':
+    desc = 'reboot the machine and perform a factory reset.'
+    parser = argparse.ArgumentParser(description=desc,
+                                     argument_default=argparse.SUPPRESS)
+    parser.add_argument('-q', '--quiet', action='store_true',
+        help='Suppress all output and don\'t ask for confirmation')
+    parser.add_argument('-D', '--disable', action='store_true',
+        help='Disable the selected stages instead of enabling them')
+
+    group = parser.add_argument_group('factory reset process stages',
+        'select which stages of the factory reset process should be performed '
+        'on the next boot. if no stage is selected all stages will be enabled')
+    for n, name, desc in STAGES:
+        group.add_argument('-'+n, '--'+name, help=desc, action='store_true')
+    args = parser.parse_args()
+
+    quiet = vars(args).pop('quiet', False)
+    disable = vars(args).pop('disable', False)
+    stages = list(vars(args).keys())
+
+    if len(stages) == 0:
+        for n, name, desc in STAGES:
+            stages.append(name)
+        if not quiet:
+            print('No stage selected, selecting all stages')
```

Comment:
I think we should change the `print` by a logger and change the level of the logger depending of the `quiet` value (`quiet is True -> level = WARNING`, `quiet is False -> level = DEBUG` )

Reply:
I also don't like how the (quiet / not quiet) logic is intertwined with the program's logic. I modeled the quiet after your original force parameter, but now I'm wondering, do we actually need a quiet or a force parameter for this program?

I'm removing `quiet` for now, and I can add it back if we decide it is necessary. I'm also adding some better logging, including saving the messages to the journal. In the future we may also want to add a metric event reporting that we are about to factory-reset the machine (and later arrange for it to be flushed to the server right away, as part of the future plans are to also reset the machine-id).

Reply:
Yes I was also to ask about the `-y` or `-f` parameter to skip confirmation, but I saw you explicitly used `--quiet` for that in the flag description, I'd prefer the `-y` or `-f` though, `--quiet` sounds more like a verbosity flag instead.

Second paragraph: sounds like a good idea.